src/lifecycle.js

```diff
@@ -366,27 +372,67 @@ function triggerLifecycle (target, component) {
  * @returns {undefined}
  */
 function initElements (elements) {
-  var elementsLen = elements.length;
-
-  for (var a = 0; a < elementsLen; a++) {
-    var element = elements[a];
-
+  var a, b, c, d, e;
+  var definitions;
+  var definitionsLength;
+  var element;
+  var elementsLength;
+  var elementsList;
+  var elementsListLength;
+  var elementsListDefinitions;
+  var elementWalker;
+  var elementWalkerCurrentNode;
+
+  elementsList = [];
+  elementsListDefinitions = [];
+  elementsLength = elements.length;
+
+  // Build a list of nodes that we will initialise.
+  for (a = 0; a < elementsLength; a++) {
+    element = elements[a];
+
+    // We screen the root node only. The rest of the nodes are screened in the
+    // tree walker.
     if (element.nodeType !== 1 || element.attributes[ATTR_IGNORE]) {
       continue;
     }
```

Comment:
I think line 374 does what you're talking about, right? I should change that slightly actually to guard against no attributes.